File: python/command.py
import json
import gameinfo

class Command(object):
    """
    creates json commands that can be sent to the server
    """
    CLIENT_TOKEN = 'client_token'
    COMM_TYPE = 'comm_type'
    NUM_PLAYERS = 'num_players'
    TEAM_NAME = 'team_name'
    MATCH_TOKEN = 'match_token'
    PASSWORD = 'password'
    GAME_MOVE = 'GameMove'
    GAME_END = 'GameEnd'
    MATCH_END = 'MatchEnd'
    
    def getMatchConnectCommand(self, team_name, match_token, team_password):
        """
        returns a json command to connect to an established match.
        """
        dict = {}
        dict[Command.COMM_TYPE] = CommType.MATCH_CONNECT
        dict[Command.TEAM_NAME] = team_name
        dict[Command.MATCH_TOKEN] = match_token
        dict[Command.PASSWORD] = team_password
        return json.dumps(dict)

class CommType(object):
    MATCH_CONNECT = 'MatchConnect'

File: c++/command.cpp
#include <iostream>
using namespace std;

#include "gameInfo.h"
#include "command.h"
#include "cJSON.h"

const string Command::Keys::CLIENT_NAME = "client_name";
const string Command::Keys::CLIENT_TOKEN = "client_token";
const string Command::Keys::COMM_TYPE = "comm_type";

string Command::getConnectCommand(string clientName)
{
  cJSON *root = cJSON_CreateObject();
  cJSON_AddItemToObject(root, "comm_type", cJSON_CreateString("MatchConnect"));
  cJSON_AddItemToObject(root, "team_name", cJSON_CreateString(clientName.c_str()));
  cJSON_AddItemToObject(root, "match_token", cJSON_CreateString(GameInfo::MATCH_TOKEN.c_str()));
  cJSON_AddItemToObject(root, "password", cJSON_CreateString(GameInfo::CLIENT_PWD.c_str()));
  return cJSON_PrintUnformatted(root);
}
